types/api.d.ts

```diff
@@ -0,0 +1,49 @@
+import {Uri} from 'vscode';
+
+declare interface BasToolkit {
+    getExtensionAPI: <T>(extensionId: string) => Promise<T>;
+    getAction: (actionId: string) => IAction | undefined;
+    getParameter: (parameterName: string) => Promise<string | undefined>;
+
+    actions: {
+        performAction: (action: IAction, options?: any) => Thenable<void>;
```

Comment:
1. actions always return `Thenable<void>` ?
2. could it be `Thenable<any>` or `Thenable<T>`?